.github/workflows/schema-dump.yml

```diff
@@ -0,0 +1,91 @@
+name: Generate and Validate Flyway Schema
+
+on:
+  pull_request:
+    types: [opened, synchronize, labeled]
+    paths:
+      - 'backend/schema/**'
+      - '.github/workflows/schema-dump.yml'
+  workflow_dispatch:
+
+jobs:
+  generate-schema:
+    runs-on: ubuntu-latest
+    
+    # Explicitly set permissions needed for committing
+    permissions:
+      contents: write
+      pull-requests: read
+
+    services:
+      postgres:
+        image: postgres:16
```

Comment:
we actually are using postgres:15 in production, so should maybe use this? but probably there are no differences.